pkg/kv/kvserver/replica_test.go

```diff
@@ -871,16 +885,26 @@ func TestReplicaRangeBoundsChecking(t *testing.T) {
 	}
 
 	gArgs := getArgs(roachpb.Key("b"))
-	_, pErr := kv.SendWrappedWith(context.Background(), tc.store, roachpb.Header{
+	ba := roachpb.BatchRequest{}
+	ba.Header = roachpb.Header{
 		RangeID: 1,
-	}, &gArgs)
+	}
+	ba.AdmissionHeader = roachpb.AdmissionHeader{}
+	ba.Add(&gArgs)
 
-	if mismatchErr, ok := pErr.GetDetail().(*roachpb.RangeKeyMismatchError); !ok {
-		t.Errorf("expected range key mismatch error: %s", pErr)
+	br, pErr := tc.store.Send(context.Background(), ba)
+
+	if pErr != nil {
+		t.Errorf("Unexpected range key mismatch error: %s", pErr)
```

Comment:
If all we're doing is checking whether pErr is non-nil, it's not necessary that this is a range key mismatch error, right? It sounds like we want `require.NotNil(t, pErr)` here instead.

Reply:
We want it to be nil here. Updated to `assertNil`.